src/pages/EnvironmentsPage/environmentsPermissions.ts

```diff
@@ -1,28 +1,35 @@
-import { LeoResourcePermissionsProvider } from 'src/analysis/Environments/Environments.models';
-import { getCreatorForCompute } from 'src/analysis/utils/resource-utils';
-import { App } from 'src/libs/ajax/leonardo/models/app-models';
+import { DecoratedComputeResource, PermissionsAndStateProvider } from 'src/analysis/Environments/Environments.models';
+import { getCreatorForCompute, isResourceDeletable } from 'src/analysis/utils/resource-utils';
+import { cromwellAppToolLabels } from 'src/analysis/utils/tool-utils';
+import { App, isApp } from 'src/libs/ajax/leonardo/models/app-models';
 import { PersistentDisk } from 'src/libs/ajax/leonardo/models/disk-models';
-import { ListRuntimeItem } from 'src/libs/ajax/leonardo/models/runtime-models';
+import { ListRuntimeItem, Runtime } from 'src/libs/ajax/leonardo/models/runtime-models';
 import { getTerraUser } from 'src/libs/state';
 
-const makePermissionsProvider = (userEmailGetter: () => string): LeoResourcePermissionsProvider => {
-  const permissionsProvider: LeoResourcePermissionsProvider = {
+const makeStateAndPermissionsProvider = (userEmailGetter: () => string): PermissionsAndStateProvider => {
+  return <PermissionsAndStateProvider>{
     canDeleteDisk: (disk: PersistentDisk) => {
       const currentUserEmail = userEmailGetter();
-      const canDelete = disk.auditInfo.creator === currentUserEmail;
-      return canDelete;
+      return disk.auditInfo.creator === currentUserEmail;
     },
     canPauseResource: (resource: App | ListRuntimeItem) => {
       const currentUserEmail = userEmailGetter();
       const creator = getCreatorForCompute(resource);
       return currentUserEmail === creator;
     },
+    canDeleteApp: (resource: DecoratedComputeResource) => {
+      if (isApp(resource)) {
+        return !Object.keys(cromwellAppToolLabels).includes(resource.appType);
+      }
```

Comment:
not sure how canDeleteApp as written is passing type checks... it has a path (if !isApp) that returns undefined, where boolean is only allowed return.

Reply:
I'm not entirely sure either, but we are doing a check in `DeleteButton.ts` where we check if the resource is an app, and if it is, we run `canDeleteApp`. If not, we pass the resource to `canDeleteResource`.